guide.md

```diff
@@ -0,0 +1,406 @@
+
+Saved draft Diesel demo with id 1
+```
+
+Unfortunately, running `show_posts` still won't display our new post, because we
+saved it as a draft. If we look back to the code in `show_posts`, we added
+`.filter(published.eq(true))`, and we had `published` default to false in our
+migration. We need to publish it! But in order to do that, we'll need to look at
+how to update an existing record. First, let's commit. The code for this demo at
+this point can be found [here][commit-no-2].
+
+Now that we've got create and read out of the way, update is actually relatively
+simple. Let's jump write into the script:
+
+```rust
+// src/bin/publish_post.rs
+extern crate diesel_demo;
+extern crate diesel;
+
+use self::diesel::prelude::*;
+use self::diesel_demo::*;
+use self::diesel_demo::models::Post;
+use std::env::args;
+
+fn main() {
+    use diesel_demo::schema::posts::dsl::{posts, published};
+
+    let id = args().nth(1).expect("publish_post requires a post id")
+        .parse::<i32>().expect("Invalid ID");
+    let connection = establish_connection();
+
+    let post = diesel::update(posts.find(id))
+        .set(published.eq(true))
+        .get_result::<Post>(&connection)
+        .expect(&format!("Unable to find post {}", id));
+    println!("Published post {}", post.title);
+}
+```
+
+That's it! Let's try it out with `cargo run --bin publish_post 1`.
+
+```
+   Compiling diesel_demo v0.1.0 (file:///Users/sean/Documents/Projects/open-source/diesel_demo)
+     Running `target/debug/publish_post 1`
+Published post Diesel demo
+```
+
+And now, finally, we can see our post with `cargo run --bin show_posts`.
+
+```
+     Running `target/debug/show_posts`
+Displaying 1 posts
+Diesel demo
+----------
+
+You know, a CLI application probably isn't the best interface for a blog demo.
+But really I just wanted a semi-simple example, where I could focus on Diesel.
+I didn't want to get bogged down in some web framework here.
+Plus I don't really like the Rust web frameworks out there. We might make a
+new one, soon.
+```
+
+We've still only covered 3 of the 4 letters of CRUD though. Let's show how to
+delete things. Sometimes we write something we really hate, and we don't have
+time to look up the ID. So let's delete based on the title, or even just some
+words in the title.
+
+```rust
+// src/bin/delete_post.rs
+extern crate diesel_demo;
+extern crate diesel;
+
+use self::diesel::prelude::*;
+use self::diesel_demo::*;
+use std::env::args;
+
+fn main() {
+    use diesel_demo::schema::posts::dsl::*;
+
+    let target = args().nth(1).expect("Expected a target to match against");
+    let pattern = format!("%{}%", target);
+
+    let connection = establish_connection();
+    let num_deleted = diesel::delete(posts.filter(title.like(pattern)))
+        .execute(&connection)
+        .expect("Error deleting posts");
+
+    println!("Deleted {} posts", num_deleted);
+}
+```
+
+We can run the script with `cargo run --bin delete_post demo` (at least with the
```

Comment:
I just like leaving unbalanced parens to make programmers feel uncomfortable (for the rest of the day